src/snowflake/connector/result_batch.py

```diff
@@ -519,18 +556,57 @@ def from_data(
 
         return new_chunk
 
-    def create_iter(
-        self, **kwargs
+    def _create_iter(
+        self, iter_unit: IterUnit
     ) -> Union[
         Iterator[Union[Dict, Exception]],
         Iterator[Union[Tuple, Exception]],
         Iterator[Table],
     ]:
-        iter_unit = kwargs.pop("iter_unit", ROW_UNIT)
+        """Create an iterator for the ResultBatch. Used by get_arrow_iter."""
         if self._local:
             return self._from_data(self._data, iter_unit)
         response = self._download()
         with TimerContextManager() as load_metric:
             loaded_data = self._load(response, iter_unit)
         self._metrics[DownloadMetrics.load.value] = load_metric.get_timing_millis()
         return loaded_data
+
+    def _get_arrow_iter(self) -> Iterator[Table]:
+        """Returns an iterator for this batch which yields a pyarrow Table"""
+        return self._create_iter(iter_unit=IterUnit.TABLE_UNIT)
+
+    def to_arrow(self) -> Optional[Table]:
+        """Returns this batch as a pyarrow Table"""
+        return next(self._get_arrow_iter(), None)
+
+    def to_pandas(self, **kwargs) -> "pandas.DataFrame":
+        """Returns this batch as a pandas DataFrame"""
+        self._check_can_use_pandas()
+        table = next(self._get_arrow_iter())
```

Comment:
The reason why I didn't add default value here is because if table = None, this function returns a dataframe. The `test_empty` in `test.arrow_pandas.py` expects no dataframes to be yielded. Hence, a `StopIteration` is what we want here. I do not have a way of making this more explicit - I'm open to your suggestions. Also, this function still returns a DataFrame if table is None (line 589), but if a StopIteration is raised, that code will never execute.

Reply:
I'm just finding that this causes a problem in Python Version > 3.6, looking in to what I can do to address this.